tmail-backend/integration-tests/imap/memory-imap-integration-tests/src/test/java/com/linagora/tmail/imap/IMAPTeamMailboxIntegrationTest.java

```diff
@@ -0,0 +1,183 @@
+                .addBinding().to(MailboxManagerClassProbe.class))
+            .overrideWith(binder -> Multibinder.newSetBinder(binder, GuiceProbe.class)
+                .addBinding().to(TeamMailboxProbe.class)))
+        .build();
+
+    @RegisterExtension
+    TestIMAPClient testIMAPClient = new TestIMAPClient();
+
+    @BeforeEach
+    void setup(GuiceJamesServer server) throws Exception {
+        server.getProbe(DataProbeImpl.class).fluent()
+            .addDomain(DOMAIN)
+            .addUser(MINISTER.asString(), MINISTER_PASSWORD)
+            .addUser(SECRETARY.asString(), SECRETARY_PASSWORD)
+            .addUser(OTHER3.asString(), OTHER3_PASSWORD);
+
+        MailboxProbeImpl mailboxProbe = server.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox(MailboxPath.inbox(MINISTER));
+        mailboxProbe.createMailbox(MailboxPath.forUser(MINISTER, "Sent"));
+        mailboxProbe.createMailbox(MailboxPath.forUser(MINISTER, "Outbox"));
+        mailboxProbe.createMailbox(MailboxPath.inbox(SECRETARY));
+        mailboxProbe.createMailbox(MailboxPath.inbox(OTHER3));
+
+        mailboxProbe.appendMessage(MINISTER.asString(),
+            MailboxPath.inbox(MINISTER),
+            MessageManager.AppendCommand.from(Message.Builder.of()
+                .setSubject("minister message subject")
+                .setBody("minister message body content 123", StandardCharsets.UTF_8)
+                .build()));
+
+        server.getProbe(DataProbeImpl.class)
+            .addAuthorizedUser(MINISTER, SECRETARY);
+
+        server.getProbe(TeamMailboxProbe.class)
+            .create(MARKETING_TEAM_MAILBOX)
+            .create(SALE_TEAM_MAILBOX)
+            .addMember(MARKETING_TEAM_MAILBOX, MINISTER)
+            .addMember(SALE_TEAM_MAILBOX, MINISTER);
+
+        server.getProbe(ACLProbeImpl.class)
+            .replaceRights(MailboxPath.inbox(OTHER3), MINISTER.asString(), MailboxACL.Rfc4314Rights.of(List.of(MailboxACL.Right.Administer, MailboxACL.Right.Lookup,
+                MailboxACL.Right.Read)));
+
+        mailboxProbe.appendMessage(MINISTER.asString(), MARKETING_TEAM_MAILBOX.mailboxPath("INBOX"),
+            MessageManager.AppendCommand.from(Message.Builder.of()
+                .setSubject("Mail in marketing team mailbox")
+                .setBody("This is content of teammailbox", StandardCharsets.UTF_8)
+                .build()));
+
+        imapPort = server.getProbe(ImapGuiceProbe.class).getImapPort();
+    }
+
+    @Test
+    void namespaceShouldReturnTeamMailboxNameSpace() throws Exception {
+        assertThat(testIMAPClient.connect(IMAP_HOST, imapPort)
+            .login(MINISTER, MINISTER_PASSWORD)
+            .sendCommand("NAMESPACE"))
+            .contains("NAMESPACE ((\"\" \".\")) ((\"#user.\" \".\")) ((\"#TeamMailbox.\" \".\"))");
+    }
+
+    @Test
+    void listShouldReturnAllTeamMailboxWhenTeamMailboxNamespace() throws Exception {
```

Comment:
Please write all variations of `LIST` similar of tests written for `#user`:

```
a0 LIST "" "*"
a1 LIST "#TeamMailbox" "*"
a2 LIST "#TeamMailbox" "marketting.*"
a3 LIST "#TeamMailbox.marbetting" "*"
a4 LIST "" "#TeamMailbox.*"
a5 LIST "" "#TeamMailbox.marketting.*"
```

Better safe than sorry...